designcompose/src/main/java/com/android/designcompose/squoosh/SquooshAnimate.kt

```diff
@@ -275,56 +279,31 @@ internal class SquooshAnimatedLayout(
 
 internal class SquooshAnimatedArc(
     target: SquooshResolvedNode,
-    private val from: Shape.Arc,
-    private val to: Shape.Arc,
+    private val from: ViewShape.VectorArc,
+    private val to: ViewShape.VectorArc,
     transition: AnimationTransition,
 ) : SquooshAnimatedItem(target, transition) {
     override fun apply(value: Float) {
         val iv = 1.0f - value
-        val arc =
-            Shape.Arc(
-                VectorArc(
-                    listOf(),
-                    listOf(),
-                    to.value.stroke_cap,
-                    from.value.start_angle_degrees * iv + to.value.start_angle_degrees * value,
-                    from.value.sweep_angle_degrees * iv + to.value.sweep_angle_degrees * value,
-                    from.value.inner_radius * iv + to.value.inner_radius * value,
-                    from.value.corner_radius * iv + to.value.corner_radius * value,
-                    to.value.is_mask,
-                )
-            )
-        // Unfortunately, the ViewData and View objects are also immutable.
+        val arc = vectorArc {
+            paths.clear()
+            strokes.clear()
+            strokeCap = to.strokeCap
+            startAngleDegrees = from.startAngleDegrees * iv + to.startAngleDegrees * value
+            sweepAngleDegrees = from.sweepAngleDegrees * iv + to.sweepAngleDegrees * value
+            innerRadius = from.innerRadius * iv + to.innerRadius * value
+            cornerRadius = from.cornerRadius * iv + to.cornerRadius * value
+            isMask = to.isMask
+        }
+
         val viewDataValue =
-            Container.Builder()
-                .apply {
-                    shape = Optional.of(ViewShape(Optional.of(arc)))
-                    children =
-                        (target.view.data.get().view_data_type.get() as ViewDataType.Container)
-                            .value
-                            .children
-                }
+            target.view.data.container
+                .toBuilder()
+                .apply { shape = viewShape { this.arc = arc } }
```

Comment:
Nit: pretty sure just `arc` is sufficient. At least it is in rust, when the variable you're assigning has the same name as the field.

Reply:
I don't quite think the kotlin apply block can achieve this. 
https://pl.kotl.in/bPOySIRSB

Anyway. I will keep the assignment.